src/mcts/search.cc

```diff
@@ -1444,16 +1445,14 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
     // Note that we want to calculate (exp(p-max_p))^(1/T) = exp((p-max_p)/T).
     p = FastExp((p - max_p) / params_.GetPolicySoftmaxTemp());
 
-    // Note that p now lies in [0, 1], so it is safe to store it in compressed
-    // format. Normalization happens later.
-    edge.edge()->SetP(p);
-    // Edge::SetP does some rounding, so only add to the total after rounding.
-    total += edge.edge()->GetP();
+    intermediate_[counter++] = p;
+    total += p;
   }
+  counter = 0;
   // Normalize P values to add up to 1.0.
   if (total > 0.0f) {
```

Comment:
This code path is pretty much always hit (it takes some truly crazy corner case to not be true IIRC), but if it isn't, edge SetP will never be called now.

Reply:
Thanks, it is fixed.